### ISSUE TITLE
Add region landmarks to Find a Store page

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13520]
**Describe the bug**

- Screen Reader users must be able to navigate by landmarks, currently, there aren't many nor labelled, therefore relevant sections on the page should be identified with meaningful landmarks names.

**Tell us the version of Spartacus**
- Library version: 4.0.0.

**To Reproduce**
1. Open any Screen Reader (VO, NVDA, JAWS).
2. Bring up the Rotor or Element List Menu.
3. Choose Landmarks view.
4. See absence of relevant landmarks.

**Expected behavior**

- Use only predefined landmarks https://wiki.wdf.sap.corp/wiki/display/spar/Landmarks
- `<cx-store-finder-search role="search" aria-label="Store Finder">`

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

@developpeurweb can you confirm please?
Should it say "region" like on product listing page?
![storeFinder](https://user-images.githubusercontent.com/45409515/146222711-5103be78-5bd4-487d-968f-5ed29a5f15e1.png)

Also, should the result area have a region too?
<img width="1308" alt="Screen Shot 2021-12-15 at 11 08 14 AM" src="https://user-images.githubusercontent.com/45409515/146222976-d52726b4-8e54-4536-b6a1-b4fa18c6f54b.png">


That's in product listing page:
<img width="1440" alt="Screen Shot 2021-12-13 at 11 52 28 AM" src="https://user-images.githubusercontent.com/45409515/146222835-ade9006e-2d80-4e55-874b-d921d47218d9.png">

**Comment by developpeurweb** [comment]
1. That's OK, `role="search"` is a search region or landmark.
2. The intent of having predefined regions or landmarks is that Screen Reader users can navigate or skip by landmarks when pressing the `D` (nvda) or `R` (jaws) keys or the `Rotor` (vo). But the landmark needs to be preloaded or pre-existent on the page to be found by the SR. Not being the case for the "All Stores" view, which gets loaded only when clicking a button. So it's OK not to have it as a landmark in there.
3. The Product Results List is OK as a landmark, because it's always there (displayed) as soon as the user lands on the page for the "Search Results" and "Category Listings" pages.

cc @KateChuen

**Comment by KateChuen** [comment]
Ok, got it. Moving it to Done then. Thanks!
